### PR TITLE
fix: upload default FileList is more than maxCount, it can't remove it

### LINKED ISSUE #47594: Upload: maxCount does not allow removing files, when files.length > maxCount
### Reproduction link

https://codesandbox.io/p/sandbox/crop-image-before-uploading-antd-5-14-2-forked-92nrsx

### Steps to reproduce

1) open https://codesandbox.io/p/sandbox/crop-image-before-uploading-antd-5-14-2-forked-92nrsx
2) try to remove image clicking by delete button
3) nothing happened

### What is expected?

image should be removed

### What is actually happening?

click by button does nothing

| Environment | Info |
| --- | --- |
| antd | 5.14.2 |
| React | 18.0.0 |
| System | macos |
| Browser | chrome |

---
there is a possibility that server can return more than maxCount images, in that case user should be able to remove them by clicking by delete button

### PR DESCRIPTION
**Pull request by Zhou-Bill**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link
fix https://github.com/ant-design/ant-design/issues/47594

### 💡 Background and solution
当初始化时的fileList >  maxCount 时，删除不起作用，无法触发onChange

解决方法： 如果当前操作的文件的`status` 为 `removed`, 可正常触发onChange

**注意： 这里如果初始的fileList 为5， maxCount 为3， 当点击删除时，当前的fileList.length = 3**

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |     fix upload fileList.length is larger than maxCount, onChange can be trigger      |
| 🇨🇳 Chinese |    修复当初始化时的fileList >  maxCount 时，删除不起作用，无法触发onChange bug       |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

**Comment by afc163** [comment]
图片

File: components/tour/__tests__/__snapshots__/index.test.tsx.snap [hunk 1]
```diff
@@ -23,31 +23,6 @@ exports[`Tour Primary 1`] = `
...
-                height="1em"
-                viewBox="64 64 896 896"
-                width="1em"
-              >
-                <path
-                  d="M799.86 166.31c.02 0 .04.02.08.06l57.69 57.7c.04.03.05.05.06.08a.12.12 0 010 .06c0 .03-.02.05-.06.09L569.93 512l287.7 287.7c.04.04.05.06.06.09a.12.12 0 010 .07c0 .02-.02.04-.06.08l-57.7 57.69c-.03.04-.05.05-.07.06a.12.12 0 01-.07 0c-.03 0-.05-.02-.09-.06L512 569.93l-287.7 287.7c-.04.04-.06.05-.09.06a.12.12 0 01-.07 0c-.02 0-.04-.02-.08-.06l-57.69-57.7c-.04-.03-.05-.05-.06-.07a.12.12 0 010-.07c0-.03.02-.05.06-.09L454.07 512l-287.7-287.7c-.04-.04-.05-.06-.06-.09a.12.12 0 010-.07c0-.02.02-.04.06-.08l57.7-57.69c.03-.04.05-.05.07-.06a.12.12 0 01.07 0c.03 0 .05.02.09.06L512 454.07l287.7-287.7c.04-.04.06-.05.09-.06a.12.12 0 01.07 0z"
-                />
-              </svg>
-            </span>
-          </button>
```
Comment (afc163) [inline components/tour/__tests__/__snapshots__/index.test.tsx.snap:50]: 不应该修改这些

Comment (Zhou-Bill) [reply components/tour/__tests__/__snapshots__/index.test.tsx.snap:50]: 嗯，我再看看

File: components/upload/Upload.tsx [hunk 2]
```diff
@@ -134,9 +134,9 @@ const InternalUpload: React.ForwardRefRenderFunction<UploadRef, UploadProps> = (
     if (event) {
       changeInfo.event = event;
     }
-
```
Comment (afc163) [inline components/upload/Upload.tsx:137]: 别删这一行。

**Review by afc163 — APPROVED** [review]

**Pull request merged**